packages/cmf/src/App.js

```diff
@@ -21,27 +21,50 @@ import { Provider } from 'react-redux';
 import history from './history';
 import RegistryProvider from './RegistryProvider';
 import UIRouter from './UIRouter';
+import onError from './onError';
+import { ErrorFeedBack } from './components';
 
 /**
  * The React component that render your app and provide CMF environment.
  * If you don't need the router, you just have to provide a children.
  * @param  {object} props { store, history }
  * @return {object} ReactElement
  */
-export default function App(props) {
-	const hist = props.history || history.get(props.store);
-	return (
-		<Provider store={props.store}>
-			<RegistryProvider>
-				{props.children || <UIRouter history={hist} loading={props.loading} />}
-			</RegistryProvider>
-		</Provider>
-	);
+export default class App extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {};
+		onError.subscribe(errors => this.setState({ error: errors[0] }));
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	componentDidCatch(error) {
```

Comment:
i am afraid that having one big componentDidCatch at the root of the application will not encourage developpers to try and catch errors early in the app tree.
When doing so could help avoid the "whiteScreenOfDeath" or here "redScreenOfDeath", maybe just a part of the state or one component did end up in an error state and catching it early will give the user the ability to keep being able to navigate away from the error.

Maybe providing a helper to still gather and log those error with documentation pushing in this path will be a good deal to ? while keeping this top level catcher for uncatched error.

Reply:
yes this is exactly the plan